trunk/source/WitnessParser/src/de/uni_freiburg/informatik/ultimate/witnessparser/yaml/IYamlProvider.java

```diff
@@ -27,13 +27,13 @@
 
 package de.uni_freiburg.informatik.ultimate.witnessparser.yaml;
 
-import com.amihaiemil.eoyaml.YamlNode;
+import java.util.Map;
 
 /**
- * Interface to transform to eo-yaml, i.e. provide a {@link YamlNode}.
+ * Interface to transform data to a map for translation to yaml.
  *
  * @author Frank Schüssele (schuessf@informatik.uni-freiburg.de)
  */
 public interface IYamlProvider {
-	YamlNode toYaml();
+	Map<String, Object> toMap();
```

Comment:
Should this really just return a map? And if so, we might want to rename the interface (though I don't like `IMapProvider` either...)

Reply:
This is currently needed for serialization, because out-of-the-box SnakeYAML can only serialize "standard" types (string, int, List, Map,...). So this method just transforms the class to a map. There is also some way for SnakeYAML to do this directly, but I have not looked to that in detail.
But yes, I don't know a good name either. I have already thought about moving this interface into `Witness`, because it is specifically for this purpose.

Reply:
I just renamed it to `IMapSerializable `, but I am unsure, if this is better...